python/huggingfaceserver/huggingfaceserver/model.py

```diff
@@ -0,0 +1,242 @@
+                self.model = AutoModelForSeq2SeqLM.from_pretrained(model_id_or_path)
+            else:
+                raise ValueError(f"Unsupported task {self.task}. Please check the supported `task` option.")
+            self.model.eval()
+            logger.info(f"successfully loaded huggingface model from path {model_id_or_path}")
+        self.ready = True
+        return self.ready
+
+    def preprocess(self, payload: Union[Dict, InferRequest], context: Dict[str, Any] = None) -> \
+            Union[BatchEncoding, InferRequest]:
+        instances = get_predict_input(payload)
+
+        # Serialize to tensor
+        if self.predictor_host:
+            inputs = self.tokenizer(
+                instances,
+                max_length=self.max_length,
+                add_special_tokens=self.add_special_tokens,
+                return_tensors=TensorType.NUMPY,
+            )
+            context["payload"] = payload
+            context["input_ids"] = inputs["input_ids"]
+            infer_inputs = []
+            for key, input_tensor in inputs.items():
+                if (not self.tensor_input_names) or (key in self.tensor_input_names):
+                    infer_input = InferInput(name=key, datatype=input_tensor.dtype,
+                                             shape=list(input_tensor.shape), data=input_tensor)
+                    infer_inputs.append(infer_input)
+            infer_request = InferRequest(infer_inputs=infer_inputs, model_name=self.name)
+            return infer_request
+        else:
+            inputs = self.tokenizer(
+                instances,
+                max_length=self.max_length,
+                add_special_tokens=self.add_special_tokens,
+                return_tensors=TensorType.PYTORCH,
+            )
+            context["payload"] = payload
+            context["input_ids"] = inputs["input_ids"]
+            return inputs
+
+    async def generate(self, generate_request: GenerateRequest, headers: Dict[str, str] = None) \
+            -> Union[GenerateResponse, AsyncIterator[Any]]:
+        parameters = generate_request.parameters
+        prompt = generate_request.text_input
+        request_id = str(uuid.uuid4())
+        if _vllm:
+            sampling_params = SamplingParams(**parameters)
+            results_generator = self.engine.generate(prompt, sampling_params=sampling_params,
+                                                     request_id=request_id)
+            return results_generator
+        else:
+            input_batch = self.tokenizer(
+                prompt,
+                max_length=self.max_length,
+                add_special_tokens=self.add_special_tokens,
+                return_tensors="pt",
+            )
+            if headers.get("streaming", "false") == "true":
+                streamer = AsyncGenerateStream(self.tokenizer)
+                generation_kwargs = dict(**input_batch, streamer=streamer)
```

Comment:
```suggestion
                generation_kwargs = dict(**input_batch, **parameters, streamer=streamer)
```
Params provided in the request need to be passed to `generate`